sympy/functions/special/delta_functions.py

```diff
@@ -195,6 +195,8 @@ def eval(cls, arg, k=0):
             return S.Zero
         if fuzzy_not(im(arg).is_zero):
             raise ValueError("Function defined only for Real Values. Complex part: %s  found in %s ." % (repr(im(arg)), repr(arg)) )
+        if arg.is_nonzero is True and arg.is_real is True:
```

Comment:
@Upabjojr reading through the [definition of nonzero](https://github.com/sympy/sympy/blob/master/sympy/core/assumptions.py#L71-L72), I think we should drop `arg.is_real`?
@rohitrango Also I think it is better to use it like
```
if arg.is_nonzero:
    do_whatever
```
Don't use the `is True` comparison. Can't find the python docs link but this will do: http://stackoverflow.com/a/4591139/3396379 .